dds/DCPS/XTypes/DynamicDataImpl.cpp

```diff
@@ -1595,6 +1608,31 @@ DDS::ReturnCode_t DynamicDataImpl::get_simple_value_string(DCPS::Value& value,
   return DDS::RETCODE_ERROR;
 }
 
+DDS::ReturnCode_t DynamicDataImpl::get_simple_value_enum(DCPS::Value& value,
+                                                         DDS::MemberId id) const
+{
+  DCPS::Value enumAsInteger(0);
+  DDS::ReturnCode_t ret = get_simple_value_primitive<DDS::Int32>(enumAsInteger, id);
```

Comment:
Similar to a previous comment, I think this needs to consider the bit bound of the enum type. We are storing the enum as an `Int8` or `Int16` or `Int32` depending on its bit bound. This call will read the `int32_` member from the corresponding `SingleValue` object for the enum, even if it was stored in the `int8_` or `int_16` member.